pkg/cluster/connection_pooler.go

```diff
@@ -285,6 +285,26 @@ func (c *Cluster) generateConnectionPoolerPodTemplate(role PostgresRole) (
 		},
 	}
 
+	var affinity *v1.Affinity
+
+	nodeAffinity := nodeAffinity(c.OpConfig.NodeReadinessLabel, spec.NodeAffinity)
+	if c.OpConfig.EnablePodAntiAffinity {
```

Comment:
can you not use `generatePodAffinity` here? Like it's done in `generatePodTemplate`? You just have to pass different labels. Or am I missing something?

Reply:
Thank you for your suggestion. I have updated